connection.js

```diff
@@ -263,9 +263,18 @@ TChannelConnection.prototype.onWriteError = function onWriteError(err) {
 TChannelConnection.prototype.onHandlerError = function onHandlerError(err) {
     var self = this;
 
-    if (err) {
-        self.resetAll(err);
+    // TODO: why is err optional?
+    if (!err) {
+        return;
+    }
+
+    if (err.isErrorFrame &&
+        err.codeName === 'Declined' &&
+        err.message.indexOf('draining:') === 0) {
+        return;
```

Comment:
I thought if a connection is draining, we want to take it out of connection pool, so it won't be re-used for new request.